File: src/main/resources/output/IFC4X3_csharp/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProcess/IfcEventType/IfcEventType.cs
public class IfcEventType : IfcTypeProcess
{
    public IfcEventTypeEnum PredefinedType { get; set; }
    public IfcEventTriggerTypeEnum EventTriggerType { get; set; }
    public IfcLabel UserDefinedEventTriggerType { get; set; }

    // === WHERE CLAUSES ===
    // CorrectEventTriggerType : (EventTriggerType <> IfcEventTriggerTypeEnum.USERDEFINED) OR ((EventTriggerType = IfcEventTriggerTypeEnum.USERDEFINED) AND EXISTS(UserDefinedEventTriggerType))
    // CorrectPredefinedType : (PredefinedType <> IfcEventTypeEnum.USERDEFINED) OR ((PredefinedType = IfcEventTypeEnum.USERDEFINED) AND EXISTS(SELF\IfcTypeProcess.ProcessType))
}

File: src/main/resources/output/IFC4X3_python/Entities/IfcRoot/IfcObjectDefinition/IfcTypeObject/IfcTypeProcess/IfcEventType/IfcEventType.py
class IfcEventType(IfcTypeProcess):
    def __init__(self):
        self.PredefinedType: IfcEventTypeEnum = None
        self.EventTriggerType: IfcEventTriggerTypeEnum = None
        self.UserDefinedEventTriggerType: IfcLabel = None

    # === WHERE CLAUSES ===
    # CorrectEventTriggerType : (EventTriggerType <> IfcEventTriggerTypeEnum.USERDEFINED) OR ((EventTriggerType = IfcEventTriggerTypeEnum.USERDEFINED) AND EXISTS(UserDefinedEventTriggerType))
    # CorrectPredefinedType : (PredefinedType <> IfcEventTypeEnum.USERDEFINED) OR ((PredefinedType = IfcEventTypeEnum.USERDEFINED) AND EXISTS(SELF\IfcTypeProcess.ProcessType))
